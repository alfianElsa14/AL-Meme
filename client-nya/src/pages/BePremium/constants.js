export const MIDTRANS_PAYMENT = 'MIDTRANS_PAYMENT'
export const SET_MIDTRANS_TOKEN = 'SET_MIDTRANS_TOKEN'
export const UPDATE_STATUS_USER = 'UPDATE_STATUS_USER'