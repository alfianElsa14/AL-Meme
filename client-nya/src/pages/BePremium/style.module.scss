@import '@styles/base';

.container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 70px;
    padding: 10px;

    @include apply-at-md-size {
        padding: 70px;
    }

    .content {
        display: flex;
        flex-direction: column;
        background-color: $color-marine-blue;
        border: 5px solid rgb(0, 83, 160);
        border-radius: 10px;
        width: 500px;
        padding: 10px 10px;
        color: $color-white;

        @include apply-at-md-size {
            padding: 30px 30px;
        }

        .title {
            font-weight: 700;
            font-size: 20px;

            @include apply-at-md-size {
                font-size: 25px;
            }
        }

        .header {
            font-size: 10px;

            @include apply-at-md-size {
                font-size: 15px;
            }
        }

        .price {
            font-weight: 700;
            font-size: 30px;

            span {
                font-size: 15px;
                font-weight: 400;
            }
        }

        .profit {
            p {
                font-size: 10px;

                @include apply-at-md-size {
                    font-size: 15px;
                }

                .check {
                    font-size: 15px;

                    @include apply-at-md-size {
                        font-size: 20px;
                    }
                }
            }
        }

        .buttonPremium {
            display: flex;
            justify-content: center;
            align-items: center;
            padding-top: 100px;
            padding-bottom: 20px;

            @include apply-at-md-size {
                padding-top: 30px;
            }

            button {
                background-color: rgb(173, 0, 0);
                color: $color-white;
                font-weight: 500;
                border: none;
                width: 200px;
                height: 35px;
                font-size: 15px;
                border-radius: 5px;
                cursor: pointer;
            }

            button:hover {
                background-color: rgb(233, 0, 0);
            }
        }


    }
}