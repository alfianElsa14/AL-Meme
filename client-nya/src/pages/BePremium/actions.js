import { MIDTRANS_PAYMENT, SET_MIDTRANS_TOKEN, UPDATE_STATUS_USER } from "./constants"

export const midtransPayment = (cbSuccess) => ({
    type: MIDTRANS_PAYMENT,
    cbSuccess
})

export const setMidtransToken = (midtransToken) => ({
    type: SET_MIDTRANS_TOKEN,
    midtransToken
})

export const updateStatusUser = (navigate) => ({
    type: UPDATE_STATUS_USER,
    navigate
})