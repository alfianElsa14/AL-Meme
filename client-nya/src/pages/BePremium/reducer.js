import { produce } from 'immer'
import { SET_MIDTRANS_TOKEN } from './constants';

export const initialState = {
    midtransToken: null
}

export const storedKey = [];

const bePremiumReducer = (state = initialState, action) =>
    produce(state, (draft) => {
        switch (action.type) {
            case SET_MIDTRANS_TOKEN:
                draft.midtransToken = action.midtransToken
                break;
            default:
                break;
        }
    })

export default bePremiumReducer;