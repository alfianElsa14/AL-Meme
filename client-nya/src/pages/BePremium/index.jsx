import React from 'react'
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';

import { midtransPayment, updateStatusUser } from './actions';

import CheckIcon from '@mui/icons-material/Check';
import classes from './style.module.scss'

function BePremium() {
    const dispatch = useDispatch();
    const navigate = useNavigate();

    const handlePaymentPremium = () => {

        dispatch(midtransPayment(() => {
            dispatch(updateStatusUser(navigate))
        }))

    }

    return (
        <div
            data-testid="bePremiumContainer"
            className={classes.container}
        >
            <div
                className={classes.content}
                data-testid="bePremiumContent"
            >
                <p 
                className={classes.title}
                data-testid="bePremiumTitle"
                ><FormattedMessage id='app_be_premium' /></p>
                <p className={classes.header}><FormattedMessage id='app_go_premium' /></p>
                <p className={classes.price}>IDR 20.000<span>/<FormattedMessage id='app_month' /></span></p>
                <div 
                data-testid="bePremiumProfit"
                className={classes.profit}
                >
                    <p><CheckIcon className={classes.check} /><FormattedMessage id='app_access_meme' /></p>
                    <p><CheckIcon className={classes.check} /><FormattedMessage id='app_create_as_many' /></p>
                    <p><CheckIcon className={classes.check} /><FormattedMessage id='app_can_edit_meme' /></p>
                </div>
                <div className={classes.buttonPremium}>
                    <button
                        data-testid="GoPremium"
                        onClick={handlePaymentPremium}>
                        Go premium
                    </button>
                </div>
            </div>
        </div>
    )
}

export default BePremium