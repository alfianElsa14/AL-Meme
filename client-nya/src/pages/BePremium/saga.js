import { midtransPayment, updateStatusUser } from "@domain/api";
import { call, put, takeLatest } from "redux-saga/effects";
import { setMidtransToken } from "./actions";
import { MIDTRANS_PAYMENT, UPDATE_STATUS_USER } from "./constants";
import { setRole, setToken, setUser } from "@containers/Client/actions";
import { setLoading } from "@containers/App/actions";
import Swal from "sweetalert2";

export function* doMidtransPayment({ cbSuccess }) {
    try {
        const result = yield call(midtransPayment)
        yield put(setMidtransToken(result.token))

        window.snap.pay(result.token, {
            onSuccess: () => {
                cbSuccess && cbSuccess()
            },
            onPending: function (result) {
                console.log('Pembayaran tertunda:', result);
            },
            onError: function (result) {
                console.log('Pembayaran gagal:', result);
            },
            onClose: function () {
                console.log('Widget ditutup tanpa menyelesaikan pembayaran');
            }
        });
    } catch (error) {
        console.log(error.response.status);
        if (error.response.status === 400) {
            const errorMessage = error.response.data.message || "Email or Password required";
            Swal.fire(errorMessage);
        } else {
            Swal.fire("Request Failed");
        }
    }
}

export function* doUpdateStatusUser({ navigate }) {
    yield put(setLoading(true))
    try {
        const response = yield call(updateStatusUser)
        yield put(setUser(response))
        yield put(setRole(response.data.role))
        yield put(setToken(response.access_token))
        yield call(navigate, '/profile')
    } catch (error) {
        console.log(error);
    }
    yield put(setLoading(false))
}

export function* bePremiumSaga() {
    yield takeLatest(MIDTRANS_PAYMENT, doMidtransPayment)
    yield takeLatest(UPDATE_STATUS_USER, doUpdateStatusUser)
}