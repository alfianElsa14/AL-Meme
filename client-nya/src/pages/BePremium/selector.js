import { createSelector } from 'reselect';
import { initialState  } from './reducer';

const selectBePremiumState = (state) => state.bePremium || initialState;

export const selectMidtransToken = createSelector(selectBePremiumState, (state) => state.midtransToken)