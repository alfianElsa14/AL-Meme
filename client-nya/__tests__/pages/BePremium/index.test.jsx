import BePremium from "@pages/BePremium";
import classes from "@pages/BePremium/style.module.scss"
import { render } from "@utils/testHelper";

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    useNavigate: () => mockNavigate,
}));

let wrapper;

beforeEach(() => {
    wrapper = render(<BePremium />)
})

describe("page be premium", () => {
    test('Should render container correctly', () => {
        const { getByTestId } = wrapper;
        const bePremiumContainer = getByTestId('bePremiumContainer');
        expect(bePremiumContainer).toHaveClass(classes.container)
        expect(bePremiumContainer).toBeInTheDocument();
    })

    test('Should render content correctly', () => {
        const { getByTestId } = wrapper;
        const bePremiumContent = getByTestId('bePremiumContent');
        expect(bePremiumContent).toHaveClass(classes.content)
        expect(bePremiumContent).toBeInTheDocument();
    })

    test('Should render title correctly', () => {
        const { getByTestId } = wrapper;
        const bePremiumTitle = getByTestId('bePremiumTitle');
        expect(bePremiumTitle).toHaveClass(classes.title)
        expect(bePremiumTitle).toBeInTheDocument();
    })
    
    test('Should render profit correctly', () => {
        const { getByTestId } = wrapper;
        const bePremiumProfit = getByTestId('bePremiumProfit');
        expect(bePremiumProfit).toHaveClass(classes.profit)
        expect(bePremiumProfit).toBeInTheDocument();
    })

    test('Should match with snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
})
