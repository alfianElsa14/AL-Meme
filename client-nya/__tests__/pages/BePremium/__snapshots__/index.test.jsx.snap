// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page be premium Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="bePremiumContainer"
      >
        <div
          class="content"
          data-testid="bePremiumContent"
        >
          <p
            class="title"
            data-testid="bePremiumTitle"
          >
            Jadi premium
          </p>
          <p
            class="header"
          >
            Jadilah premium dan buat meme sebanyak yang Anda inginkan.
          </p>
          <p
            class="price"
          >
            IDR 20.000
            <span>
              /
              bulan
            </span>
          </p>
          <div
            class="profit"
            data-testid="bePremiumProfit"
          >
            <p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              Akses 50+ Template Meme Eksklusif!
            </p>
            <p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              Buat meme sebanyak yang Anda inginkan tanpa batas
            </p>
            <p>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              Anda dapat mengedit meme yang Anda buat
            </p>
          </div>
          <div
            class="buttonPremium"
          >
            <button
              data-testid="GoPremium"
            >
              Go premium
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="bePremiumContainer"
    >
      <div
        class="content"
        data-testid="bePremiumContent"
      >
        <p
          class="title"
          data-testid="bePremiumTitle"
        >
          Jadi premium
        </p>
        <p
          class="header"
        >
          Jadilah premium dan buat meme sebanyak yang Anda inginkan.
        </p>
        <p
          class="price"
        >
          IDR 20.000
          <span>
            /
            bulan
          </span>
        </p>
        <div
          class="profit"
          data-testid="bePremiumProfit"
        >
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
            Akses 50+ Template Meme Eksklusif!
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
            Buat meme sebanyak yang Anda inginkan tanpa batas
          </p>
          <p>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium check css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
              />
            </svg>
            Anda dapat mengedit meme yang Anda buat
          </p>
        </div>
        <div
          class="buttonPremium"
        >
          <button
            data-testid="GoPremium"
          >
            Go premium
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
